src/model/walletConnect.ts

```diff
@@ -0,0 +1,404 @@
+import AsyncStorage from '@react-native-community/async-storage';
+import analytics from '@segment/analytics-react-native';
+import { captureException } from '@sentry/react-native';
+import WalletConnectClient, { CLIENT_EVENTS } from '@walletconnectv2/client';
+import { Reason, SessionTypes } from '@walletconnectv2/types';
+import lang from 'i18n-js';
+import { clone } from 'lodash';
+import { Alert, InteractionManager, Linking } from 'react-native';
+import { enableActionsOnReadOnlyWallet } from '../config/debug';
+import { isSigningMethod } from '../utils/signingMethods';
+import { sendRpcCall } from '@rainbow-me/handlers/web3';
+import { getDappMetadata } from '@rainbow-me/helpers/dappNameHandler';
+import { convertHexToString, delay } from '@rainbow-me/helpers/utilities';
+import walletConnectApprovalSheetTypes from '@rainbow-me/helpers/walletConnectApprovalSheetTypes';
+import { walletConnectSupportedChainIds } from '@rainbow-me/helpers/walletConnectNetworks';
+import walletTypes from '@rainbow-me/helpers/walletTypes';
+import { Navigation } from '@rainbow-me/navigation';
+import {
+  addRequestToApproveV2,
+  WC_VERSION_2,
+} from '@rainbow-me/redux/requests';
+import {
+  RAINBOW_METADATA,
+  saveWalletConnectV2Sessions,
+  WALLETCONNECT_V2_UPDATE_SESSIONS,
+} from '@rainbow-me/redux/walletconnect';
+import Routes from '@rainbow-me/routes';
+import { logger, watchingAlert } from '@rainbow-me/utils';
+
+const wcTrack = (
+  event: string,
+  metadata: { name: string; url: string },
+  opts?: object
+) =>
+  analytics.track(event, {
+    dappName: metadata?.name || metadata?.url,
+    dappUrl: metadata?.url,
+    version: WC_VERSION_2,
+    ...opts,
+  });
+
+const SUPPORTED_MAIN_CHAINS = [
+  'eip155:1',
+  'eip155:10',
+  'eip155:137',
+  'eip155:42161',
+];
```

Comment:
I think it's time to add the chainId inside `networkInfo.js` and then you can filter out all the networks with testnet === true